spring-cloud-stream-binder-jms-common/src/main/java/org/springframework/cloud/stream/binder/jms/utils/DestinationNameResolver.java

```diff
@@ -1,77 +0,0 @@
-/*
- *  Copyright 2002-2017 the original author or authors.
- *
- *  Licensed under the Apache License, Version 2.0 (the "License");
- *  you may not use this file except in compliance with the License.
- *  You may obtain a copy of the License at
- *
- *        http://www.apache.org/licenses/LICENSE-2.0
- *
- *  Unless required by applicable law or agreed to in writing, software
- *  distributed under the License is distributed on an "AS IS" BASIS,
- *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *  See the License for the specific language governing permissions and
- *  limitations under the License.
- */
-
-package org.springframework.cloud.stream.binder.jms.utils;
-
-import java.util.ArrayList;
-import java.util.Collection;
-
-import org.springframework.cloud.stream.binder.ConsumerProperties;
-import org.springframework.cloud.stream.binder.ProducerProperties;
-import org.springframework.util.StringUtils;
-
-/**
- * Component responsible of building up the name of a specific queue given some context
- *
- * @author José Carlos Valero
- * @author Donovan Muller
- * @since 1.1
- */
-public class DestinationNameResolver {
```

Comment:
One issue with removing these classes is that they can be used in other JMS-based binders - such as the IBM MQ binder: https://github.com/spring-cloud/spring-cloud-stream-binder-ibm-mq/blob/master/src/main/java/org/springframework/cloud/stream/binder/jms/ibmmq/IBMMQProvisioningProvider.java#L55. Since topic/queue naming is a general pattern for the JMS binder(s), would it be make sense to have the code extracted as a utility?